internal/api/connectors/inventory/inventory.go

```diff
@@ -0,0 +1,192 @@
+		Fields:   &fields,
+	}
+}
+
+func NewInventoryClientWithHttpRequestDoer(cfg *viper.Viper, doer HttpRequestDoer) InventoryConnector {
+	client := &ClientWithResponses{
+		ClientInterface: &Client{
+			Server: fmt.Sprintf("%s://%s:%d%s", cfg.GetString("inventory.connector.scheme"), cfg.GetString("inventory.connector.host"), cfg.GetInt("inventory.connector.port"), basePath),
+			Client: utils.NewMeasuredHttpRequestDoer(doer, "inventory", "GetHostConnectionDetails"),
+			RequestEditor: func(ctx context.Context, req *http.Request) error {
+				req.Header.Set(constants.HeaderRequestId, request_id.GetReqID(ctx))
+
+				if identity, ok := ctx.Value(constants.HeaderIdentity).(string); ok {
+					req.Header.Set(constants.HeaderIdentity, identity)
+				}
+
+				return nil
+			},
+		},
+	}
+
+	return &inventoryConnectorImpl{
+		client: client,
+	}
+}
+
+func NewInventoryClient(cfg *viper.Viper) InventoryConnector {
+	httpClient := http.Client{
+		Timeout: time.Duration(cfg.GetInt64("inventory.connector.timeout") * int64(time.Second)),
+	}
+
+	return NewInventoryClientWithHttpRequestDoer(cfg, &httpClient)
+}
+
+func (this *inventoryConnectorImpl) getHostDetails(
+	ctx context.Context,
+	IDs []string,
+	orderBy string,
+	orderHow string,
+	limit int,
+	offset int,
+) (details []HostOut, err error) {
+
+	params := createHostGetHostByIdParams(orderBy, orderHow)
+
+	response, err := this.client.ApiHostGetHostByIdWithResponse(ctx, IDs, params)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if response.JSON200 == nil {
+		return nil, utils.UnexpectedResponse(response.HTTPResponse)
+	}
+
+	return response.JSON200.Results, err
+}
+
+func (this *inventoryConnectorImpl) getSystemProfileDetails(
+	ctx context.Context,
+	IDs []string,
+	orderBy string,
+	orderHow string,
+	limit int,
+	offset int,
+) (details map[string]HostSystemProfileOut, err error) {
+
+	params := createHostGetHostSystemProfileByIdParams(orderBy, orderHow)
+
+	response, err := this.client.ApiHostGetHostSystemProfileByIdWithResponse(ctx, IDs, params)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if response.JSON200 == nil {
+		return nil, utils.UnexpectedResponse(response.HTTPResponse)
+	}
+
+	formatedResults := keySystemProfileResults(response.JSON200.Results)
+
+	return formatedResults, nil
+}
+
+func (this *inventoryConnectorImpl) GetHostConnectionDetails(ctx context.Context, IDs []string, order_how string, order_by string, limit int, offset int) (details []HostDetails, err error) {
+
+	hostResults, err := this.getHostDetails(ctx, IDs, order_how, order_by, limit, offset)
+
+	if err != nil {
+		return nil, err
+	}
+
+	systemProfileResults, err := this.getSystemProfileDetails(ctx, IDs, order_by, order_how, limit, offset)
+
+	if err != nil {
+		return nil, err
+	}
+
+	hostConnectionDetails := make([]HostDetails, len(IDs))
+	for i, host := range hostResults {
+		_, IDPresent := systemProfileResults[*host.Id]
+		if IDPresent {
+			satelliteFacts := getSatelliteFacts(host.Facts)
```

Comment:
I think the `getSatelliteFacts()` and the construction of the `HostDetails{}` instance belongs outside of the `if IDPresent` check.  You might just want to fill in the system profile based info if the system profile data exists.

Maybe I'm looking at that wrong.